test/test_apply_rhs.jl

```diff
@@ -0,0 +1,75 @@
+function test_apply_rhs()
+    grid = generate_grid(Quadrilateral, (20, 20))
+    dim = 2
+    ip = Lagrange{dim,RefCube,1}()
+    qr = QuadratureRule{dim,RefCube}(2)
+    cellvalues = CellScalarValues(qr, ip)
+    
+    dh = DofHandler(grid)
+    push!(dh, :u, 1)
+    close!(dh)
+    
+    K = create_sparsity_pattern(dh)
+    
+    ch = ConstraintHandler(dh)
+    
+    ∂Ω = union(getfaceset.((grid,), ["left", "right", "top", "bottom"])...)
+    
+    dbc = Dirichlet(:u, ∂Ω, (x, t) -> 0)
+    add!(ch, dbc);
+    
+    close!(ch)
+    update!(ch, 0.0);
+    
+    function doassemble(
+        cellvalues::CellScalarValues{dim},
+        K::SparseMatrixCSC,
+        dh::DofHandler,
+    ) where {dim}
+    
```

Comment:
I would either make this `doassemble!(K, f, cellvalues, dh)` and create K and f outside or `doassemble(cellvalues, dh)` and create both K and f inside.